src/TTreeWidget.cpp

```diff
@@ -255,17 +257,113 @@ void TTreeWidget::rowsInserted(const QModelIndex& parent, int start, int end)
 
         int newParentID = parent.data(Qt::UserRole).toInt();
         if (mIsTriggerTree) {
-            mpHost->getTriggerUnit()->reParentTrigger(mChildID, mOldParentID, newParentID, parentPosition, childPosition);
+            if (mOldParentID == newParentID) {
+                return;
+            }
+            if (!mParentItem) {
+                return;
+            }
+            const int m_ParentItemID = mParentItem->data(0, Qt::UserRole).toInt();
```

Comment:
As it's a local variable, I wouldn't prefix it with `m_`, just `parentItemID` here and in the other cases would be enough.